### ISSUE TITLE
The calendar widget needs to support `widget_template`

### ISSUE THREAD

**Issue by Elv13** [awesomeWM/awesome#2862]
Ref 
https://www.reddit.com/r/awesomewm/comments/cpqhhy/problem_with_adding_buttons_to_wiboxwidgetcalendar/

The current API is a mess. It's that way because at the time, the `widget_template` concept didn't exist. @psychon pointed this out back then, but I preferred to merge the widget because I didn't have time to implement the API and the I don't like to have otherwise good PRs from new contributors stalled for too long. This discourage them from contributing again. But now I have to pay the price. I create this ticket to "block" v4.4 (it should have been done in v4.3, but I forgot).

**Comment by actionless** [comment]
@Elv13 how this became a release-blocker?

**Comment by Elv13** [comment]
Right, there is no way this block anything. I moved the milestone. Originally, I wanted "all"  composite widgets (calendar, layoutbox, etc) to move to support templates, but with nobody working on it, it ain't happening.
